lib/models/NotificationConfiguration.js

```diff
@@ -289,21 +289,23 @@ class NotificationConfiguration {
      */
     static validateConfig(config) {
         assert.strictEqual(typeof config, 'object');
-        assert.strictEqual(Array.isArray(config.queueConfig), true);
-        config.queueConfig.forEach(q => {
-            const { events, queueArn, filterRules, id } = q;
-            assert.strictEqual(Array.isArray(events), true);
-            events.forEach(e => assert.strictEqual(typeof e, 'string'));
-            assert.strictEqual(typeof queueArn, 'string');
-            if (filterRules) {
-                assert.strictEqual(Array.isArray(filterRules), true);
-                filterRules.forEach(f => {
-                    assert.strictEqual(typeof f.name, 'string');
-                    assert.strictEqual(typeof f.value, 'string');
-                });
-            }
-            assert.strictEqual(typeof id, 'string');
-        });
+        if (config.queueConfig) {
+            assert.strictEqual(Array.isArray(config.queueConfig), true);
+            config.queueConfig.forEach(q => {
```

Comment:
Not sure this matters, but I just want to compare two assertions and mention that even without asserting something you are effectively writing assertions in certain cases.
These two lines contain two asserts first the strict equal and second that queueConfig contains a forEach function.
The reason for this is that assert just takes some values or conditions and throws an exception if some condition is not met. Similarly trying to call a function on a variable that has no function of that name will also throw an exception because that condition was not met.

I also want to compare the output of these two assertions as sometimes not asserting can give more information in logs than writing an assert.

The first creates an exception that basically says `assertion error: false !== true` if the queue config is not an array. A message that is not too helpful without the line number that gives us the assertion.
The second "hidden" assert gives an error similar to `Uncaught TypeError: config.queueConfig.forEach is not a function`. In my opinion, the second error gives more valuable debug information without needing extra verbose assert messages. And because any error results in an invalid configuration it should work identically.

tl;dr the assert is not required as the forEach method is only found on arrays anyway.

Reply:
i removed all the array asserts before `.forEach` uses